### ISSUE TITLE
Source-to-Devfile-to-Workspace 

### ISSUE THREAD

**Issue by gestrem** [eclipse/che#23080]
### Is your enhancement related to a problem? Please describe

When I want to use Devspaces from a git repo, I have to reuse a devFile, customize it (add git url,components, UDI,runtimes etc)  and add it to my git repo. Finally I manage the devfile on my own which leads to a lack consistency from my corp / devteam standards (runtimes versions, CVE patches, build params...) 

### Describe the solution you'd like

Provide devfiles for specific stacks (like samples provided on the   'Create workspace' view ) and be able to customize it with a git repo url, creds, ports, runtimes version. Then the devFile is set and the workspace is launched with user requirements that fits with corporate / dev tools .. 
This workflow is very close to the source-to-image flow proposed inside OCP.
Finally, it's a new way to bring governance inside developer onboarding. Which follows the move / market trends according to Developer Hub (Backstage) and RHTSSC...

### Describe alternatives you've considered

Follow the source-to-image flow to "assemble" and customize the devfile with the git source repo.

**Comment by tolusha** [comment]
@gestrem 
Do I understand correctly, you would like to have a view on a dashboard to paste devfile and create a workspace from it?

**Comment by gestrem** [comment]
@tolusha not exactly. The user choose a stack (defined as a devfile behind the scene) , add its source code repo location. Then some options are suggested to set runtimes versions, port, buidl / test command and the workspace starts.

**Comment by ibuziuk** [comment]
@gestrem please, provide more details, motivation, use-cases diagrams to the description
